src/map.rs

```diff
@@ -2367,6 +2371,63 @@ where
     }
 }
 
+impl<'de, K, V> Deserialize<'de> for HashMap<K, V, crate::DefaultHashBuilder>
```

Comment:
I wonder if we might want to also include `S` as generic here `where S: Default`, rather than forcing `DefaultHashBuilder`.

Reply:
That would be great. The only issue is that we wouldn't be able to preallocate the map, but that may not be an issue for most use cases (?).

Reply:
Why not? You can just use `HashMap::with_capacity_and_hasher`?

Reply:
I see. You're right! I could use `with_hasher`/`with_capacity_and_hasher` for preallocating the map.

Should we implement `HashMap::with_capacity`, `HashMap::new`and `Default` for `HashMap<K, V, S>` where `S: Default` though? That would make the API a bit more flexible (and I can't see why that wouldn't work).

Reply:
Ah, no, `with_capacity` and `new` are specifically _not_ generic over `S`, because it leads to the user having to explicitly give `let x: HashMap<K, V> = HashMap::new` everywhere to opt into the default. Having `impl Default` be generic over `S` makes sense to me though.

Reply:
I could work on that - having `impl Default` be generic over `S` - after we're finished with this.